src/v/coproc/tests/utils/coprocessor.h

```diff
@@ -200,7 +200,7 @@ struct two_way_split_copro : public coprocessor {
 
 inline model::topic
 to_materialized_topic(const model::topic& src, const model::topic& dest) {
-    return model::topic(ssx::sformat("{}.${}$", src(), dest()));
+    return model::topic(ssx::sformat("{}._{}_", src(), dest()));
```

Comment:
Should we also update topic name validation logic ?

Reply:
Underscores are valid chars

Reply:
I mean do we have some special handling for dollar signs ?

Reply:
We never did. The way we got around it was just by going straight to v/storage when creating materaizlied logs, and never going through v/kafka or v/cluster, where the validation checks are located. In this way we bypassed the validation, but if a user wanted to create a log with that schema, they would be blocked.